tests/DeviceServiceTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dot\tests;
+
+use DeviceDetector\DeviceDetector;
+use Dot\UserAgentSniffer\Data\DeviceData;
+use Dot\UserAgentSniffer\Service\DeviceService;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class DeviceServiceTest extends TestCase
+{
+    private const TYPE_SMARTPHONE = 'smartphone';
+    private const TYPE_BROWSER    = 'browser';
+
+    private DeviceService $subject;
+
+    private DeviceDetector|MockObject $deviceDetectorMock;
+
+    private DeviceData|MockObject $deviceDataMock;
```

Comment:
The IDE reports that both these properties (`$deviceDetectorMock` and `$deviceDataMock`) can be replaced with a local variable, by removing them from the class property list and adding them locally in the only method that uses them, `testGetDetailsReturnsDeviceData`.
Then don't forget to `Add PHPDoc comment with @throws tags`, as the IDE will suggest.